tools/keepalived/keepalived/check/libipvs.c

```diff
@@ -704,10 +703,17 @@ ipvs_get_service(ipvs_service_t *hint, lcoreid_t cid)
 	len_rcv = sizeof(*dpvs_svc_rcv);
 	memset(&dpvs_svc, 0, len);
 
-	dpvs_app_ptr = &dpvs_app;
-	memset(dpvs_app_ptr, 0, sizeof(dpvs_app));
-	IPVS_2_DPVS(dpvs_app_ptr, hint);
-	memcpy(&dpvs_svc, dpvs_app_ptr, sizeof(dpvs_app));
+	dpvs_svc.user.af = hint->af;
+	dpvs_svc.user.proto = hint->user.protocol;
+	memcpy(&dpvs_svc.user.addr, &hint->nf_addr, sizeof(dpvs_svc.user.addr));
+	dpvs_svc.user.port = hint->user.port;
+	dpvs_svc.user.fwmark = hint->user.fwmark;
+	dpvs_svc.user.flags = hint->user.flags;
+	dpvs_svc.user.netmask = hint->user.netmask;
+	snprintf(dpvs_svc.user.srange, sizeof(dpvs_svc.user.srange), "%s", hint->user.srange);
+	snprintf(dpvs_svc.user.drange, sizeof(dpvs_svc.user.drange), "%s", hint->user.drange);
+	snprintf(dpvs_svc.user.iifname, sizeof(dpvs_svc.user.iifname), "%s", hint->user.iifname);
+	snprintf(dpvs_svc.user.oifname, sizeof(dpvs_svc.user.oifname), "%s", hint->user.oifname);
```

Comment:
I think the macro `IPVS_2_DPVS` also supports type `dpvs_service_entry_t`. Please make use of it to simplify the codes.

Reply:
already simplify the codes